dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/EventDataValuesValidationHook.java

```diff
@@ -172,24 +167,28 @@ private void validateDataValueDataElementIsConnectedToProgramStage( ValidationEr
         }
     }
 
-    private void validateFileNotAlreadyAssigned( ValidationErrorReporter reporter, Event event, DataValue dataValue,
-        DataElement dataElement )
+    private TrackerErrorCode getTrackerErrorCode( ErrorMessage errorMessage )
     {
-        if ( dataValue == null || dataValue.getValue() == null )
-        {
-            return;
-        }
-
-        boolean isFile = dataElement.getValueType() != null && dataElement.getValueType().isFile();
-        if ( !isFile )
+        switch ( errorMessage.getErrorCode() )
         {
-            return;
+        case E2027:
```

Comment:
if these are the same mappings we've already declared you could add an EnumMap in a shared place for these hooks to use. If not, just ignore this comment 😋